Feature/parametric module

stefan-voelk opened the pull request:
## Before you get started
- [x] 🙋‍♀️ Create an issue to discuss what you are going to do!

---

## Description
- Closes #84 

Adding a **parametric module** considering **climate optimized operations**. Post-processing methodology after MSc thesis Saleh Walie (2025) and parameters derived from Castino (2024).

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation change

## How has this been tested?
Unit tests added in `tests/parametric_test.py`

- [x] `TestAdaptCo2Emissions`
- [x] `TestAdaptRf`

Furthermore, end-to-end tests have been executed locally with different settings (parametric module enabled/disabled, default/customized parameters).

**Test configuration**:
* Operating system: Windows 11 Enterprise
* Environment: environment_dev.yaml

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any changed dependencies have been added or removed correctly
- [ ] main branch: I have updated the CHANGELOG
- [ ] main branch: I have updated the version numbering in `__about__.py` according to the [semantic versioning scheme](https://semver.org/)

ahsawa committed:
Add parametric scenario module

Add initial version of parametric module (work in progress), docs and tests still pending.

Related to #84

stefan-voelk committed:
Merge branch 'main' into feature/parametric-module

stefan-voelk committed:
Update main.py, example.toml, parametric_scenario.py, read_conifg.py

Restore parametric commands in main.py which have been deleted during merge,
update DEFAULT_CONFIG in read_config.py,
refactor code

stefan-voelk committed:
Fix main.py

Remove double "TOTAL" identifier for species_sub (PMO)

stefan-voelk committed:
Rename parametric module, add documentation and tests

stefan-voelk committed:
Update main.py

Fix output parameters of adapt_rf function.

stefan-voelk committed:
Update parametric.py, parametric.rst, main.py

Change default parameters in RATIO_DIC_D to the values stated in Castino (2024) and Saleh Walie (2025).
Add documentation, after section 4.5 "Development of the post-processing strategy" in Saleh Walie (2025).
Add warning in main.py due to emissions of non-CO2 species.

stefan-voelk committed:
Update example.toml

disabling parametric module and customized parameters in example config

stefan-voelk committed:
Merge branch 'main' into feature/parametric-module

stefan-voelk committed:
Add oac.parametric.rst

Enabling API documentation

stefan-voelk committed:
Update main.py and parametric.py

Update warning when parametric module is activated. Add SWV to default parameters.

liammegill reviewed (changes requested):
Please have a look at the comments. I think we need to figure out a way of making it clear that the intermediate emissions/concentrations values are not valid if the parametric module is enabled. Otherwise the code itself is essentially good to go.

stefan-voelk committed:
Update main.py, read_config.py

New line added before if statement as requested in PR software review.
Typo in read_config.py fixed.

stefan-voelk committed:
Delete .DS_Store

File was a left-over from MacOS

stefan-voelk committed:
Update main.py, write_output.py

Filter output if parametric module is enabled.

stefan-voelk committed:
Update write_output.py, write_output_test.py

Add function filter_parametric_output() which prevents non-CO2 emissions and concentrations to be written to the output.

Add unit tests for that function. Fix pytest config fixture in TestWriteOutputDictToNetcdf.

stefan-voelk committed:
Fix two warnings in write_output_test.py

Change deprecated ds.dims to ds.sizes since the former one will be changed in the future.

liammegill committed:
Merge branch 'main' into feature/parametric-module

liammegill reviewed (approved):
Looks good to me. Thanks for your work on this @stefan-voelk!
